### ISSUE TITLE
Change the response when get we an HTTP request and expect HTTPS

### ISSUE THREAD

**Issue by ttmc** [bigchaindb/nginx_3scale#5]
If we are expecting HTTPS (i.e. SSL/TLS), but NGINX gets an HTTP request (i.e. not HTTPS), NGINX should respond as outlined here:

https://tools.ietf.org/html/rfc2817#section-4.2

I looked into how we might make NGINX respond to HTTP requests (on port 80) with HTTP response code 426 Upgrade Required, plus those two special headers. I think it might be something like this (but I'm not an NGINX wizard):

```text
server {
    listen 80 default_server;
    listen [::]:80 default_server;
    # TODO: edit the server_name below:
    server_name api.example.com;
    location / {
        add_header Upgrade "TLS/1.2, HTTP/1.1";
        add_header Connection "Upgrade";
        return 426;
    }
}
```

The same change should be made to `k8s/nginx/container/nginx.conf.template` over in the bigchaindb/bigchaindb repository.

**Comment by krish7919** [comment]
The current response is:

```
400 Bad Request

The plain HTTP request was sent to HTTPS port
openresty/1.11.2.2
```

I will park this for now and get back to it once we are done with the clustering and security.

**Comment by ttmc** [comment]
Note: the 400 Bad Request response code is what you get if you go to

http://example.com:443

but if you go to

http://example.com   (i.e. on port 80)

then there is no response at all.

**Comment by ttmc** [comment]
Fixed by #8
